python/setup.py

```diff
@@ -0,0 +1,18 @@
+from setuptools import setup, Distribution
+
+
+class BinaryDistribution(Distribution):
+    def has_ext_modules(foo):
+        return True
+
+
+setup(
+    name="pyrecordio",
+    version="0.0.5",
+    description="recordio file format support",
+    url="https://github.com/wangkuiyi/recordio",
+    license="Apache 2.0",
+    packages=["recordio"],
+    package_data={"recordio": ["librecordio.so"]},
```

Comment:
It looks to me that this Python script should call Go toolchain to build the shared library, and we can remove build.sh?

Reply:
Interesting idea. This file is not called user, but by 'pip wheel' command. I can dig a bit deeper to see how the file is used.

Reply:
When pip is invoked, it creates a temporary directory and in that directory it creates a symbol links the current directory and run setup.py there. It is possible but messy to find the go directory and compile go code.
How about adding a Makefile in root directory to build python package?

Reply:
I don't understand your explanation about why we cannot make setup.py build the .so file.

Basing on my limited knowledge about pip package building, we run 

```
python setup.py sdist
```

to create the .tar.gz package to be registered to PyPI.

I don't understand why would it be messy if there is a symbolic link pointing to "the current directory" where the Go source files reside. And I don't understand why would a Makefile solve this "problem"?

Reply:
the "current directory“ is the directory containing "setup.py", not the go directory.
```
python setup sdist
```
According to the [doc](https://docs.python.org/3.7/distutils/sourcedist.html) is for distributing pure python code, no compilation is done.

the makefile idea is to use makefile to replace the build.sh